### PR TITLE
[INLONG-3828][Sort] Enhanced the node filter capabilities

### LINKED ISSUE #3828: [Feature][Sort] Enhanced the node filter capabilities
### Description

Enhanced the node filter capabilities.Increase the filter strategy, one is to remain the data that meets the filter, and the other is to remove the data that meets the filter.

### Use case

_No response_

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by yunqingmoswu**
### Title Name: [INLONG-3828][Sort] Enhanced the node filter capabilities

Fixes #3828 

### Motivation

*Explain here the context, and why you're making that change. What is the problem you're trying to solve?*

### Modifications

*Describe the modifications you've done.*

### Verifying this change

*(Please pick either of the following options)*

- [ ] This change is a trivial rework/code cleanup without any test coverage.

- [ ] This change is already covered by existing tests, such as:
  *(please describe tests)*

- [ ] This change added tests and can be verified as follows:

  *(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a follow-up issue for adding the documentation

### CODE REVIEW

File: inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/sort/util/FilterFunctionUtils.java [hunk 1]
```diff
@@ -91,6 +92,32 @@ public static List<FilterFunction> createFilterFunctions(FilterDefinition filter
         return filterFunctions;
     }
 
+    public static FilterStrategy parseFilterStrategy(TransformResponse transformResponse) {
```
Comment (healchow) [inline inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/sort/util/FilterFunctionUtils.java:95]: Please add java doc for static method, thanks.

Comment (yunqingmoswu) [reply inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/sort/util/FilterFunctionUtils.java:101]: done

**Review by healchow — APPROVED** [review]
LGTM

File: inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/node/load/HiveLoadNode.java [hunk 2]
```diff
@@ -88,7 +90,7 @@ public HiveLoadNode(@JsonProperty("id") String id,
             @JsonProperty("hiveVersion") String hiveVersion,
             @JsonProperty("hadoopConfDir") String hadoopConfDir,
             @JsonProperty("parFields") List<FieldInfo> partitionFields) {
-        super(id, name, fields, fieldRelationShips, filters, sinkParallelism, properties);
+        super(id, name, fields, fieldRelationShips, filters, filterStrategy, sinkParallelism, properties);
```
Comment (thexiay) [inline inlong-sort/sort-common/src/main/java/org/apache/inlong/sort/protocol/node/load/HiveLoadNode.java:93]: recommand every load node,every extract node  have a builder mode, so every time we add filed in 'LoadNode' or 'ExtractNode',We can avoid modifying the call site of the constructor call.
Better open an issue to follow it.

**Review by EMsnap — APPROVED** [review]

**Review by thexiay — APPROVED** [review]

**Review by gong — APPROVED** [review]

**Pull request merged**